src/prefect/agent/fargate/agent.py

```diff
@@ -144,20 +75,70 @@ def __init__(
             region_name=region_name,
         )
 
-        # Look for default subnets with `MapPublicIpOnLaunch` disabled
-        if not self.subnets:
-            self.logger.debug("No subnets provided, finding defaults")
-            ec2 = boto3_client(
-                "ec2",
-                aws_access_key_id=aws_access_key_id,
-                aws_secret_access_key=aws_secret_access_key,
-                aws_session_token=aws_session_token,
-                region_name=region_name,
-            )
-            for subnet in ec2.describe_subnets()["Subnets"]:
-                if not subnet.get("MapPublicIpOnLaunch"):
-                    self.subnets.append(subnet.get("SubnetId"))
-                    self.logger.debug("Subnet {} found".format(subnet.get("SubnetId")))
+    def _parse_kwargs(self, user_kwargs: dict) -> tuple:
+        """
+        Parse the kwargs passed in and separate them out for `register_task_definition`
+        and `run_task`. This is required because boto3 does not allow extra kwargs
+        and if they are provided it will raise botocore.exceptions.ParamValidationError.
+
+        Args:
+            - user_kwargs (dict): The kwargs passed to the initialization of the environment
+
+        Returns:
+            tuple: a tuple of two dictionaries (task_definition_kwargs, task_run_kwargs)
+        """
+        definition_kwarg_list = [
+            "taskRoleArn",
+            "executionRoleArn",
+            "volumes",
+            "placementConstraints",
+            "cpu",
+            "memory",
+            "tags",
+            "pidMode",
+            "ipcMode",
+            "proxyConfiguration",
+            "inferenceAccelerators",
+        ]
+
+        run_kwarg_list = [
+            "cluster",
+            "count",
+            "startedBy",
+            "group",
+            "placementConstraints",
```

Comment:
Are there any edge cases / bugs that could be introduced with `placementConstraints`, since it equally applies to runs and definitions?  E.g., is it possible a user would want to apply some setting here to _only_ the definition or _only_ the run?